cloudinit/config/cc_apt_configure.py

```diff
@@ -6,225 +6,364 @@
+                'disable_suites': {
+                    'type': 'array',
+                    'items': {
+                        'type': 'string'
+                    },
+                    'uniqueItems': True,
+                    'description': dedent("""\
+                        Entries in the sources list can be disabled using
+                         ``disable_suites``, which takes a list of suites
+                         to be disabled. If the string ``$RELEASE`` is
+                         present in a suite in the ``disable_suites`` list,
+                         it will be replaced with the release name. If a
+                         suite specified in ``disable_suites`` is not
+                         present in ``sources.list`` it will be ignored.
+                         For convenience, several aliases are provided for
+                         ``disable_suites``:
+
+                             ``updates`` => ``$RELEASE-updates``,
+                             ``backports`` => ``$RELEASE-backports``.
+                             ``security`` => ``$RELEASE-security``,
+                             ``proposed`` => ``$RELEASE-proposed``,
+                             ``release`` => ``$RELEASE``.
+
+                        When a suite is disabled using ``disable_suites``,
+                        its entry in ``sources.list`` is not deleted; it
+                        is just commented out.""")
+                },
+                'primary': {
+                    **mirror_property,
+                    'description': dedent("""\
+                        The primary and security archive mirrors can
+                        be specified using the ``primary`` and
+                         ``security`` keys, respectively Both the
+                         ``primary`` and ``security`` keys take a list
+                        of configs, allowing mirrors to be specified
+                        on a per-architecture basis. Each config is a
+                        dictionary which must have an entry for
+                         ``arches``, specifying which architectures
+                        that config entry is for. The keyword
+                         ``default`` applies to any architecture not
+                        explicitly listed. The mirror url can be specified
+                        with the ``uri`` key, or a list of mirrors to
+                        check can be provided in order, with the first
+                        mirror that can be resolved being selected. This
+                        allows the same configuration to be used in
+                        different environment, with different hosts used
+                        for a local apt mirror. If no mirror is provided
+                        by ``uri`` or ``search``, ``search_dns`` may be
+                        used to search for dns names in the format
+                         ``<distro>-mirror`` in each of the following:
+
```

Comment:
@lucasmoura So, we'll have to chat about this Monday post standup.
There is one simple thing we need to fix, but a 2nd more complex thing we'll have to chat about Monday.
1. Each item listed below ``<distro>-mirror`` in each of the following is actually supposed to be a bulleted list with leading hyphens in order to properly render in sphinx documentation for readthedocs.

to check your readthedocs content generation locally:
tox -e doc
open a browser to something like file:///home/<YOUR_USER>/src/cloud-init/doc/rtd_html/topics/modules.html#apt-configure

Compare that against https://cloudinit.readthedocs.io/en/latest/topics/modules.html#apt-configure
to see significant unexpected discrepancies.

2.  There is whitespace damage when rendering docs via the CLI `python3 -m cloudinit.cmd.main devel schema -d cc_apt_configure`  due to https://github.com/canonical/cloud-init/blob/master/cloudinit/config/schema.py#L321

Originally, when schema coverage was simple, I had the expectation that property descriptions were going to be simple and possibly one concise line of detail which could be collapsed into a single line. With this case we have a need for pre-formatted content rendering to avoid whitespace damage.

We will likely have to make the whitespace collapsing smarter (here)[https://github.com/canonical/cloud-init/blob/master/cloudinit/config/schema.py#L321] to allow for use to provide a <pre> format tag or something in the places where we don't want to collapse it all.

Reply:
Yes, I agree with you on that. I can start working on this without an issue, but should I address this on this PR or in a separate one ?